apps/web/app/api/v1/auth.ts

```diff
@@ -0,0 +1,37 @@
+import { authOptions } from "@/app/api/auth/[...nextauth]/authOptions";
+import { getServerSession } from "next-auth";
+import { getApiKeyFromKey } from "@formbricks/lib/services/apiKey";
+import { getEnvironmentBySession } from "@formbricks/lib/services/environment";
+import { TAuthentication, TAuthenticationApiKey, TAuthenticationSession } from "@formbricks/types/v1/auth";
+
+export async function getAuthentication(request: Request): Promise<TAuthentication | null> {
+  const apiKey = request.headers.get("x-api-key");
+  const session = await getServerSession(authOptions);
```

Comment:
please move `getServerSession` in an `else` case after checking the apiKey as this call is expensive and isn't needed when we have an api key :-)